staging/src/k8s.io/client-go/plugin/pkg/client/auth/gcp/gcp.go

```diff
@@ -113,7 +113,16 @@ type gcpAuthProvider struct {
 	persister   restclient.AuthProviderConfigPersister
 }
 
+var warnOnce sync.Once
+
 func newGCPAuthProvider(_ string, gcpConfig map[string]string, persister restclient.AuthProviderConfigPersister) (restclient.AuthProvider, error) {
+	// deprecated in v1.22, remove in v1.25
+	// this should be updated to use klog.Warningf in v1.24 to more actively warn consumers
+	warnOnce.Do(func() {
+		klog.V(1).Infof(`WARNING: in-tree gcp auth plugin is now deprecated.
+Please use the gcloud kubectl/client-go credential plugin instead.`)
```

Comment:
@mikedanese are you happy with this verbiage?

Reply:
I'll defer to mike on content, but I'd suggest a single line if possible

Reply:
also, in other deprecation warnings, we've included the "from" version and the targeted removal version, e.g.
```
WARNING: the gcp auth plugin is deprecated in v1.22+, unavailable in v1.25+; use ___  instead
```

Reply:
> I'll defer to mike on content, but I'd suggest a single line if possible

The docs link forced me to use multiple lines to keep the length sane.

Updated to include the from and target versions.